UPGRADE-4.0.md

```diff
@@ -0,0 +1,243 @@
+# Upgrade from v3.x to v4.x
+
+Please consult the `examples` directory for reference example configurations. If you find a bug, please open an issue with supporting configuration to reproduce.
+
+## List of backwards incompatible changes
+
+- Minimum supported version of Terraform AWS provider updated to v4.17 to support latest resources
+- Minimum supported version of Terraform updated to v1.0
+- `create` variable added to control whether all resources are created or not. This means that all resources now use the zeroth index `[0]` in the resource name
+
+## Additional changes
+
+### Added
+
+- Support for generating a random password for the `master_password`
+- `aws_redshift_snapshot_schedule` resource including the variables and outputs used to support it
+- `aws_redshift_snapshot_schedule_association` resource including the variables and outputs used to support it
+- `aws_redshift_scheduled_action` resource including support for creating the IAM role and policies plus the associated variables and outputs to support
+- `aws_redshift_usage_limit` resource including the variables and outputs used to support it
+- `aws_redshift_authentication_profile` resource including the variables and outputs used to support it
+- `aws_redshift_cluster_iam_roles` resource including the variables and outputs used to support it
+
+### Modified
+
+- `number_of_nodes` default value of `3` changed to `1`
+- `cluster_version` default value of `"1.0"` changed to `null`
+- `cluster_node_type` default value of `dc2.large` added
```

Comment:
Should `cluster_node_type` be up to a user to specify and avoid default values?

Reply:
yes - not sure what I was thinking there. removed